src/daft-sql/src/functions.rs

```diff
@@ -90,6 +91,20 @@ pub struct SQLFunctions {
     map: HashMap<String, Arc<dyn SQLFunction>>,
 }
 
+pub(crate) struct SQLFunctionArguments {
+    pub unnamed: HashMap<usize, ExprRef>,
+    pub named: HashMap<String, ExprRef>,
+}
+
+impl SQLFunctionArguments {
+    pub fn get_unnamed(&self, idx: usize) -> Option<&ExprRef> {
+        self.unnamed.get(&idx)
+    }
+    pub fn get_named(&self, name: &str) -> Option<&ExprRef> {
+        self.named.get(name)
+    }
+}
```

Comment:
Having this struct is nicer than having to do the validation in each function, but there's no way to check for any unsupported arguments that were passed in using this right now. They will just be ignored, which may not be the best behavior. 

Perhaps we can add something to `ScalarUDF` that specifies the expected unnamed and named inputs, so that the input definition is only specified in one place. I'm not entirely sure what that would look like yet but just a thought.

Reply:
@kevinzwang great suggestion. I added some parameters to `plan_function_args` to enforce this.

```rs
// will error if there are named arguments that are not 'mode' or 'on_error'
// and will error if there are > 0 positional arguments. 
let args = planner.plan_function_args(args, &["mode", "on_error"], 0)?;
```